tcms/report/views.py

```diff
@@ -584,41 +619,25 @@ def get_context_data(self, **kwargs):
 
         if form.is_valid():
             test_case_runs = self.get_case_runs(form)
-            status_names = TestCaseRunStatus.get_names()
-            priority_values = dict(Priority.objects.values_list('pk', 'value'))
-
-            testers_ids, assignees_ids = self._get_testers_assignees_ids(
-                test_case_runs)
-            testers = self.get_related_users(testers_ids)
-            assignees = self.get_related_users(assignees_ids)
 
             data['test_case_runs_count'] = len(test_case_runs)
-            data['test_case_runs'] = self.walk_case_runs(test_case_runs,
-                                                         status_names,
-                                                         priority_values,
-                                                         testers,
-                                                         assignees)
+            data['test_case_runs'] = self.walk_case_runs(test_case_runs)
         else:
             data['form_errors'] = form.errors
 
         return data
 
-    def _get_testers_assignees_ids(self, case_runs):
-        testers_ids = set()
-        assignees_ids = set()
-        for case_run in case_runs:
-            pk = case_run.tested_by_id
-            if pk:
-                testers_ids.add(case_run.tested_by_id)
-            pk = case_run.assignee_id
-            if pk:
-                assignees_ids.add(case_run.assignee_id)
-        return list(testers_ids), list(assignees_ids)
-
-    def walk_case_runs(self, case_runs, status_names, priority_values,
-                       testers, assignees):
+    def walk_case_runs(self, test_case_runs):
         # todo: this is the same method as in testruns/views.py
```

Comment:
See this todo comment ? See if you can't merge the two methods together, maybe as helper functions.

Reply:
Merge them as one function? Where should i store it?

Reply:
Maybe in `tcms/utils.py`, give it a try and we'll see how it goes.